client/app/scripts/superdesk-authoring/tests/authoring_spec.js

```diff
@@ -1030,30 +1027,60 @@ describe('authoring actions', function() {
             allowedActions(itemActions, ['new_take', 'save', 'edit', 'duplicate', 'view', 'spike',
                     'mark_item', 'package_item', 'multi_edit', 'publish', 'send']);
         }));
+});
 
-    describe('authoring workspace controller', function() {
-        var ctrl, scope;
+describe('authoring workspace', function() {
 
-        beforeEach(inject(function($controller, $rootScope) {
-            scope = $rootScope.$new();
-            scope.flags = {};
-            ctrl = $controller('AuthoringWorkspace', {$scope: scope});
-        }));
+    beforeEach(module('superdesk.authoring'));
 
-        it('can edit item', inject(function() {
-            var item = {};
+    it('can edit item', inject(function(superdeskFlags, authoringWorkspace) {
+        var item = {};
 
-            expect(scope.flags.authoring).toBeFalsy();
+        expect(superdeskFlags.flags.authoring).toBeFalsy();
 
-            ctrl.edit(item);
-            expect(ctrl.item).toBe(item);
-            expect(ctrl.getItem()).toBe(item);
-            expect(scope.flags.authoring).toBeTruthy();
+        authoringWorkspace.edit(item);
+        expect(authoringWorkspace.item).toBe(item);
+        expect(authoringWorkspace.getItem()).toBe(item);
+        expect(superdeskFlags.flags.authoring).toBeTruthy();
+        expect(authoringWorkspace.readonly).toBe(false);
 
-            ctrl.close();
-            expect(ctrl.item).toBe(null);
-            expect(ctrl.getItem()).toBe(null);
-            expect(scope.flags.authoring).toBeFalsy();
-        }));
-    });
+        authoringWorkspace.close();
+        expect(authoringWorkspace.item).toBe(null);
+        expect(authoringWorkspace.getItem()).toBe(null);
+        expect(superdeskFlags.flags.authoring).toBeFalsy();
+
+        authoringWorkspace.edit(item, true);
+        expect(authoringWorkspace.readonly).toBe(true);
+    }));
```

Comment:
Seems like this test case is trying to test to much (`edit()`, `close()`, then `edit()` again), IMO it should probably be split to a couple of smaller independent test cases.